### PR TITLE
Bug 2030758 - Changes the Llm abstraction to receieve an optional sys…

### PR DESCRIPTION
**Pull request by boek**
[try](https://treeherder.mozilla.org/jobs?repo=try&landoInstance=lando-prod&landoCommitID=191364)

### CODE REVIEW

**Review by MatthewTighe — APPROVED** [review]

File: mobile/android/android-components/components/lib/llm-mlpa/src/main/java/mozilla/components/lib/llm/mlpa/MlpaLlm.kt [hunk 1]
```diff
@@ -26,8 +26,9 @@ internal class MlpaLlm(
 internal val Prompt.asRequest
     get() = Request(
         model = ModelID.mozSummarization,
-        messages = listOf(
-            Message.user(value),
-        ),
+        messages = buildList {
+            systemPrompt?.let { add(Message.system(it)) }
+            add(Message.user(userPrompt))
+        },
```
Comment (MatthewTighe) [inline mobile/android/android-components/components/lib/llm-mlpa/src/main/java/mozilla/components/lib/llm/mlpa/MlpaLlm.kt:32]: Should we add a test case for the null branch?
